lib/stdlib/src/erl_abstract_code.erl

```diff
@@ -0,0 +1,17 @@
+-module(erl_abstract_code).
+-export([debug_info/4]).
+
+debug_info(_Format, _Module, {none,_CompilerOpts}, _Opts) ->
+    {error, missing};
+debug_info(erlang_v1, _Module, {AbstrCode,_CompilerOpts}, _Opts) ->
+    {ok, AbstrCode};
+debug_info(core_v1, _Module, {AbstrCode,CompilerOpts}, Opts) ->
+    CoreOpts = [to_core, binary, return_errors] ++ CompilerOpts ++ Opts,
+    try compile:noenv_forms(AbstrCode, CoreOpts) of
+	{ok, _, Core} -> {ok, Core};
```

Comment:
If the option `return_warnings` is included in Opts, then the return value will be a tuple with four elements, the last element being the list of warnings. That caused Dialyzer to crash in our daily builds. It seems that sometimes Dialyzer passes the `return_warnings` option and sometimes it doesn't.

It is probably easiest to add a new clause:

    {ok, _, Core, _} -> {ok, Core};